backend/src/main/java/com/bakdata/conquery/models/query/ManagedQuery.java

```diff
@@ -86,17 +90,31 @@ public Stream<EntityResult> streamResults(OptionalLong maybeLimit) {
 
 	@Override
 	public synchronized long resultRowCount() {
-		if (lastResultCount == null) {
+		Long count = updatedResultCount();
+		if (count == null) {
 			throw new IllegalStateException("Result row count is unknown, because the query has not yet finished.");
 		}
+		return count;
+	}
+
+	/**
+	 * @implNote this is a workaround for a race condition, because the execution state is handled externally by
+	 * the {@link ExecutionManager}. The manager might have updated the lastResultCount in the meantime.
+	 * Here is a mismatch because the {@link ExecutionStatus} for an execution might be constructed for an
+	 * "outdated" object (the {@link ExecutionManager} updated the stored execution asynchronously).
+	 */
+	public Long updatedResultCount() {
+		if (lastResultCount == null) {
+			lastResultCount = ((ManagedQuery) getId().resolve()).getLastResultCount();
+		}
 		return lastResultCount;
 	}
```

Comment:
@awildturtok ich habe hier das Problem, dass ich von einem ManagedQuery Object einen Status generieren lassen möchte. Während ich das mache, wird die Query fertig und im ExecutionManager steht sie auf DONE.
Der ExecutionManager führt dann finish auf einem ManagedQuery Object aus (das potentiell vom Cache geladen wurde), was aber nicht dieses Object ist.

So kann es sein, dass der Status zwar DONE ist aber `lastResultCount` in diesem Object noch auf `null` steht, weil der Cache nicht dieses Object, sondern ein neues zurückgegeben hat.